crypto/ct/ct_sct.c

```diff
@@ -317,12 +317,11 @@ int SCT_set_source(SCT *sct, sct_source_t source)
 
 int SCT_LIST_set_source(const STACK_OF(SCT) *scts, sct_source_t source)
 {
-    int i, ret = 1;
+    int i, ret = 0;
 
     for (i = 0; i < sk_SCT_num(scts); ++i) {
-        ret = SCT_set_source(sk_SCT_value(scts, i), source);
-        if (ret != 1)
-            break;
+        if (SCT_set_source(sk_SCT_value(scts, i), source))
```

Comment:
Then, similarly to SCT_LIST_set_source, wouldn't it be better to have the caller decide how they want to treat unknown SCTs, rather than skip them and say "some were skipped"? If you wanted to know if this method actually did _anything_, then you'd rather awkwardly have to count the SCTs with set logs first...